aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -171,38 +172,190 @@ Tensor ger(const Tensor& self, const Tensor& vec2) {
   return result;
 }
 
-Tensor addbmm_cpu(const Tensor& self, const Tensor& batch1, const Tensor& batch2, Scalar beta, Scalar alpha) {
-  Tensor b_self;
-  std::tie(b_self) = expand_size(self, {batch1.size(1), batch2.size(2)}, "addbmm");
-  return legacy::cpu::_th_addbmm(b_self, batch1, batch2, beta, alpha);
+static void addmm_impl_cpu_(
+    Tensor &result, const Tensor &self, Tensor m1, Tensor m2, Scalar beta, Scalar alpha) {
+  TORCH_CHECK(self.dim() == 2, "input must be a matrix");
+  TORCH_CHECK(m1.dim() == 2, "m1 must be a matrix");
+  TORCH_CHECK(m2.dim() == 2, "m2 must be a matrix");
```

Comment:
This is a slight pessimization over the old error checking code, which would tell you what the dimension of input/m1/m2 were (btw, we should use names consistent with the Python documentation, which are `input`, `mat1` and `mat2`